File: 11-container-with-most-water/11-container-with-most-water.scala
object Solution {
  def maxArea(height: Array[Int]): Int = {
    var l = 0
    var r = height.length - 1
    var res = 0

    while (l < r) {
      res = res max ((height(l) min height(r)) * (r - l))
      if (height(r) >= height(l)) {
        l += 1
      } else {
        r -= 1
      }
    }

    res
  }
}

File: 11-container-with-most-water/11-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int res = 0;
        
        int l = 0;
        int r = height.length - 1;
        
        while(l < r) {
            res = Math.max(Math.min(height[l], height[r]) * (r - l), res);
            if (height[l] >= height[r]) r--;
            else l++;
        }
        
        return res;
    }
}